### ISSUE TITLE
Add-on no longer available as Firefox Add-On

### ISSUE THREAD

**Issue by danmattera** [vladikoff/netflix-1080p-firefox#28]
Just wondering if there's any word on when/if it'll be back up?

**Comment by vladikoff** [comment]
Yeah it got banned for including the external netflix component. we shall see how we can bring it back

**Comment by Edszx** [comment]
Hey, @vladikoff, thank you very much for the update, but, were can we find the addon packaged for the time being while it's not on the mozilla repository?

**Comment by ItsNickBarry** [comment]
You can install the extension manually.  I think this is the process:
1. Clone or download the repository
2. Navigate to `about:addons`
3. In the settings menu, choose "Install Add-on From File..."
4. Choose the `manifest.json` file

**Comment by Edszx** [comment]
> 
> 
> You can install the extension manually. I think this is the process:
> 
>     1. Clone or download the repository
> 
>     2. Navigate to `about:addons`
> 
>     3. In the settings menu, choose "Install Add-on From File..."
> 
>     4. Choose the `manifest.json` file

Doesn't work, Firefox says that the addon is corrupted, probably the directory structure is not correct.

**Comment by ItsNickBarry** [comment]
Apparently it needs to be signed even if installed manually from a file.  [See here](https://stackoverflow.com/questions/31952727/how-can-i-disable-signature-checking-for-firefox-add-ons) for an alternative.  It might also need to be compiled with [web-ext](https://www.npmjs.com/package/web-ext), but I'm not sure.

You could also load the addon temporarily (removed when the browser session ends) by navigating to `about:debugging` and choosing "Load Temporary Add-on..." (select the `manifest.json` file).

**Comment by bamboozel** [comment]
about:debugging "trick" works :). Thank you.

**Comment by kulmegil** [comment]
Mozilla (like the Pilate guy from the bible) sure does like to wash their hands. They will chicken and remove everything that could cause them legal disturbance, but are nice enough to at least leave backdoor open.

You should look into e.g. [Bypass Paywalls](https://github.com/iamadamdev/bypass-paywalls-firefox) also been removed from AMO. They host it from github exclusively and auto-updates are still supported. Ofc each .xpi need to be signed.

**Comment by progandy** [comment]
For linux, you might be able to put an unsigned .xpi in ```/usr/share/mozilla/extensions```

https://github.com/mozilla/gecko-dev/blob/master/toolkit/mozapps/extensions/internal/XPIDatabase.jsm#L361
>  // On UNIX platforms except OSX, an additional location for system
> // add-ons exists in /usr/{lib,share}/mozilla/extensions. Add-ons
> // installed there do not require signing.

**Comment by Louirj1** [comment]
> Mozilla (like the Pilate guy from the bible) sure does like to wash their hands. They will chicken and remove everything that could cause them legal disturbance, but are nice enough to at least leave backdoor open.
> 
> You should look into [Bypass Paywalls](https://github.com/iamadamdev/bypass-paywalls-firefox), that has also been removed from AMO. They host it from github exclusively and auto-updates are still supported. Ofc each .xpi needs to be signed.

You think you can help me install it not sure where to start

**Comment by kulmegil** [comment]
@Louirj1 
The ball is in @vladikoff court to pick best course of action. He still have the option to provide updates off AMO.

In the meantime just install [1.8 signed](https://1n.pm/1NhnV) version. It still works fine for me.
(If You have trouble installing just save .xpi on disk, and drag into fx window).

**Comment by Multipanda** [comment]
@kulmegil Thanks, if you have access to it, could you also upload 1.7? (that version should have 5.1 support)

Hopefully once vladikoff has come to a solution, and 1.9 can be installed, that version will either have an option like the chrome version, or just have 5.1 enabled by default, something I've been sorely missing.

**Comment by kulmegil** [comment]
@Multipanda no, from all my profile backups I could only dig v1.8

But if You really want it, I got the code and sign it myself [v1.7](https://1n.pm/A0QH3). Keep in mind that since version is not signed by Vlad it will be installed parallel to existing extension! Remove / disable the other one.

**Comment by Multipanda** [comment]
@kulmegil Thanks I appreciate it, 5.1 still seems to work just fine, now I don't have to switch to Chrome :-)

**Comment by Javier-DarthPalpatine** [comment]
> 
> 
> Yeah it got banned for including the external netflix component. we shall see how we can bring it back

Hi Vlad!
I have my copy of version 1.9 signed.  Can I post it here? 

Tks for this amazing addon!
Javier

**Comment by vladikoff** [comment]
Sure, post yours for now. I can publish another version later

**Comment by kld87** [comment]
@Javier-DarthPalpatine @vladikoff is the 1.9 signed posted somewhere?

**Comment by Javier-DarthPalpatine** [comment]
@vladikoff  @kld87 

**Here is version 1.9 !!**:
[Netflix_1080p_v1.9_for_Fx.zip](https://github.com/vladikoff/netflix-1080p-firefox/files/3438656/Netflix_1080p_v1.9_for_Fx.zip)   (unzip file to get the xpi)

Vlad, could you please upload/post any updated version (here or anywhere) even if no signed?  
_(It's rather easy to install unsigned addons in released Firefox versions by searching a little)_
It'd be much appreciated!

Thanks!
J

**Comment by mikeloeven** [comment]
Is it possible to convert this to a user-script format for use with tamper monkey

**Comment by daishishi** [comment]
Anyone else having trouble with the add-on manually installed? On my system i am locked on a SD quality(540p). The debugging window show the Error Message: 'Could not establish connection. Receiving end does not exist.'
![Captura de tela de 2019-08-03 22-53-20](https://user-images.githubusercontent.com/7233354/62418564-6596f700-b642-11e9-9359-3aafead667ef.png)
Thanks for all the work.

**Comment by Strongaloff** [comment]
> Anyone else having trouble with the add-on manually installed? On my system i am locked on a SD quality(540p). The debugging window show the Error Message: 'Could not establish connection. Receiving end does not exist.'
> ![Captura de tela de 2019-08-03 22-53-20](https://user-images.githubusercontent.com/7233354/62418564-6596f700-b642-11e9-9359-3aafead667ef.png)
> Thanks for all the work.

Also here. Using Manjaro + Firefox Dev and with the addon enabled (manually installed), the cap is set at 540p.

Tried deleting data/cache + reinstalled browser + reinstalled addon, after that it worked just one time on 1080p then switched to 540p and capped.

**Comment by GrayBoltWolf** [comment]
> Anyone else having trouble with the add-on manually installed? On my system i am locked on a SD quality(540p). The debugging window show the Error Message: 'Could not establish connection. Receiving end does not exist.'
> ![Captura de tela de 2019-08-03 22-53-20](https://user-images.githubusercontent.com/7233354/62418564-6596f700-b642-11e9-9359-3aafead667ef.png)
> Thanks for all the work.

Same here. Locked to low bitrates:
![image](https://user-images.githubusercontent.com/4554196/62495147-1b795700-b7a3-11e9-930b-b697908affdd.png)

**Comment by daishishi** [comment]
> 
> 
> Anyone else having trouble with the add-on manually installed? On my system i am locked on a SD quality(540p). The debugging window show the Error Message: 'Could not establish connection. Receiving end does not exist.'
> ![Captura de tela de 2019-08-03 22-53-20](https://user-images.githubusercontent.com/7233354/62418564-6596f700-b642-11e9-9359-3aafead667ef.png)
> Thanks for all the work.

I'm using Firefox 68.0.1 (64-bits), on both Ubuntu 18.04 and Windows 8.1.
I realized that if you install the add-on it will work the first time, then if you reload the page or change the show / episode it locks on a 540p resolution on all the subsequent attempts. Uninstalling the Add-on return the resolution to 720p.

**Comment by Frydac** [comment]
Using Manjaro + Firefox 68.0.1 and manually installed the signed 1.9 version from Javier-DarthPalpatine https://github.com/vladikoff/netflix-1080p-firefox/issues/28#issuecomment-515717459
Seems to work for me consistently.

**Comment by GrayBoltWolf** [comment]
> Using Manjaro + Firefox 68.0.1 and manually installed the signed 1.9 version from Javier-DarthPalpatine's [post](https://github.com/vladikoff/netflix-1080p-firefox/issues/28#issuecomment-515717459)
> Seems to work for me consistently.

Seems to be working here now as well.

![image](https://user-images.githubusercontent.com/4554196/63228311-f6cba900-c1be-11e9-9a80-4eefb9dd0496.png)


Debian Testing, Firefox 68.8.0

**Comment by Multipanda** [comment]
Just a shame that the 1.9 doesn't offer 5.1 sound, though at least in that respect the 1.7 still works for me. If I have to choose between a higher bitrate or 5.1 sound, the latter wins out.
I would just prefer having both :-)

**Comment by daishishi** [comment]
> Using Manjaro + Firefox 68.0.1 and manually installed the signed 1.9 version from Javier-DarthPalpatine's [post](https://github.com/vladikoff/netflix-1080p-firefox/issues/28#issuecomment-515717459)
> Seems to work for me consistently.

After your reply, and updating the firefox to the Version 68.0.2 I tested the Add-on 1.9 version signed. It is working fine for me now on Ubuntu 18.04.3 LTS, i didn't tested on Windows 8.1 thou.

**Comment by bullfroggie** [comment]
Signed 1.9v works flawlessly in Firefox 68.0.02 - Fedora

**Comment by Multipanda** [comment]
At least for me there's no 5.1 sound in 1.9 (Firefox 68.0.2 64-bit)
Netflix only shows 2.0 audio tracks

If I remove 1.9 and go back to 1.7 I have access to 5.1 tracks again.

**Comment by hashar** [comment]
I would recommend to just build and sign the extension yourself. It is not that complicated. In short I did:

Create an account on https://addons.mozilla.org/

Generate a key + secret: https://addons.mozilla.org/developers/addon/api/key/ , which would give something like:

Key: user:1234567:890
Secret: long_hexadecimal_string

Which you would need when signing the extension using `web-ext sign`.


git clone https://github.com/vladikoff/netflix-1080p-firefox/
cd netflix-1080p-firefox
npm install
./node_modules/.bin/web-ext sign -s src --api-key "Key from addons.mozilla.org" --api-secret "Secret from addons.mozilla.org"
firefox web-ext-artifacts/netflix_1080p-1.8-an+fx.xpi

If all went well, Firefox will prompt for some authorization, and you can watch the Test Pattern video to confirm you have 1080p: https://www.netflix.com/watch/80018585

**Comment by SF97** [comment]
> 
> 
> I would recommend to just build and sign the extension yourself. It is not that complicated. In short I did:
> 
> Create an account on https://addons.mozilla.org/
> 
> Generate a key + secret: https://addons.mozilla.org/developers/addon/api/key/ , which would give something like:
> 
> Key: user:1234567:890
> Secret: long_hexadecimal_string
> 
> Which you would need when signing the extension using `web-ext sign`.
> 
> ```
> git clone https://github.com/vladikoff/netflix-1080p-firefox/
> cd netflix-1080p-firefox
> npm install
> ./node_modules/.bin/web-ext sign -s src --api-key "Key from addons.mozilla.org" --api-secret "Secret from addons.mozilla.org"
> firefox web-ext-artifacts/netflix_1080p-1.8-an+fx.xpi
> ```
> 
> If all went well, Firefox will prompt for some authorization, and you can watch the Test Pattern video to confirm you have 1080p: https://www.netflix.com/watch/80018585

Running the web-ext sign command outputted the following: Your add-on failed validation and could not be signed
FAIL

**Comment by fweimer** [comment]
Would it be possible to download the problematic files from the Netflix site and patch them locally, so that they do not have to be shipped in the add-on directly?

**Comment by Multipanda** [comment]
Anyone willing to create a signed version with okradonkey's two pull requests that adds the 5.1 selector?

**Comment by chanssl** [comment]
@hashar Hi, I try to sign the add-on, but it's fail to sign. 

> Server response: You do not own this addon. (status: 403)

Did you konwn how to slove it? thank you.

**Comment by Frydac** [comment]
@chanssl as stated in another issue ( https://github.com/vladikoff/netflix-1080p-firefox/issues/33 ), there is a new 1080p plugin in the Firefox store that seems to be up to date and working:
https://addons.mozilla.org/en-US/firefox/addon/netflix-1080p-firefox
https://github.com/TheGoddessInari/netflix-1080p-firefox

**Comment by chanssl** [comment]
> 
> 
> @chanssl as stated in another issue ( #33 ), there is a new 1080p plugin in the Firefox store that seems to be up to date and working:
> https://addons.mozilla.org/en-US/firefox/addon/netflix-1080p-firefox
> https://github.com/TheGoddessInari/netflix-1080p-firefox

Thank you! :-)

**Comment by Edu4rdSHL** [comment]
Seems that Firefox is able to play 1080p in Linux when using native Wayland support.

OS: ArchLinux
Firefox version: 74

![2020-04-12-110841-sechacklabs](https://user-images.githubusercontent.com/32582878/79073954-1fa13100-7caf-11ea-882d-37612b3f2b8f.png)

**Comment by PedroHLC** [comment]
@Edu4rdSHL can't reproduce with Nightly(`76.0a1 (2020-03-26)`)+Wayland
Can you post your about:support in a gist? (use 'Copy text to clipboard' button).

**Comment by Edu4rdSHL** [comment]
Yes, it's here: https://gist.github.com/Edu4rdSHL/57a51283b30501e888f6bd0e79a0d584

Correction: the Firefox version is 75, not 74 as I mentioned above,

**Comment by PedroHLC** [comment]
:thinking: you don't even have dmabuf-vaapi enabled. But it seems like Firefox benchmarked your hardware as being capable of decoding 233fps using VP9, maybe this is interfering in MediaCapabilities (just my assumption)

Just curious, try enabling dmabuf-vaapi (make sure it's working on your arch first, using `vainfo` from `libva-utils`), forcing enable webgpu, webrender and webrender compositor.

**Comment by Edu4rdSHL** [comment]
Sincerely I do not want to start testing/changing things in my OS now, I am busy, sorry.

I have done the **EDIT2** step, here's the link: https://gist.github.com/Edu4rdSHL/ce2c161deab1321fca8edc729fca6d68

`intel-media-driver` and `libva` are already installed.

**Comment by d-sfounis** [comment]
> I would recommend to just build and sign the extension yourself. It is not that complicated. In short I did:
> 
> Create an account on https://addons.mozilla.org/
> 
> Generate a key + secret: https://addons.mozilla.org/developers/addon/api/key/ , which would give something like:
> 
> Key: user:1234567:890
> Secret: long_hexadecimal_string
> 
> Which you would need when signing the extension using `web-ext sign`.
> 
> ```
> git clone https://github.com/vladikoff/netflix-1080p-firefox/
> cd netflix-1080p-firefox
> npm install
> ./node_modules/.bin/web-ext sign -s src --api-key "Key from addons.mozilla.org" --api-secret "Secret from addons.mozilla.org"
> firefox web-ext-artifacts/netflix_1080p-1.8-an+fx.xpi
> ```
> 
> If all went well, Firefox will prompt for some authorization, and you can watch the Test Pattern video to confirm you have 1080p: https://www.netflix.com/watch/80018585

This doesn't work, shows a "you don't own this addon" error after running `web-ext` :(

**Comment by thepwrtank18** [comment]
```
Error: Validation took too long to complete; last status: {"guid":"{f9a7af99-e1c5-481b-9682-41f44c378e5e}","active":false,"automated_signing":true,"url":"https://addons.mozilla.org/api/v3/addons/%7Bf9a7af99-e1c5-481b-9682-41f44c378e5e%7D/versions/1.8/uploads/b7223bd60625490ab5af52e970796b40/","files":[{"download_url":"https://addons.mozilla.org/api/v3/file/4082018/5353020e601946c2bade-1.8.zip","hash":"sha256:9efb70f9ba5de50100a349cf409f3c178c882f75322c1b4fe5482752308330ef","signed":false}],"passed_review":false,"pk":"b7223bd60625490ab5af52e970796b40","...
    at Timeout._onTimeout (C:\Users\[...]\Desktop\netflix-1080p-firefox-1.9.0\node_modules\sign-addon\dist\webpack:\src\amo-client.js:232:16)
    at listOnTimeout (node:internal/timers:569:17)
    at processTimers (node:internal/timers:512:7)
```

**Comment by NeatNit** [comment]
I've uploaded the Mozilla-signed extension here: https://github.com/vladikoff/netflix-1080p-firefox/issues/63#issuecomment-1470154520
